lib/storage/data/external/AwsClient.js

```diff
@@ -237,44 +264,39 @@ class AwsClient {
                     .customizeDescription('Error returned from ' +
                   `${this.type}: ${err.message}`),
                 );
-            }
-            return callback();
-        });
+            });
     }
 
     healthcheck(location, callback) {
         const awsResp = {};
-        this._client.headBucket({ Bucket: this._awsBucketName },
-            err => {
-            /* eslint-disable no-param-reassign */
-                if (err) {
+        this._client.send(new HeadObjectCommand({ Bucket: this._awsBucketName }))
+        .catch(err => {
                     awsResp[location] = { error: err, external: true };
                     return callback(null, awsResp);
-                }
+        }).then(() => {
```

Comment:
Maybe not the order of promise we want, as after calling the callback in the .catch, we will execute the .then?